tests/test_node.py

```diff
@@ -67,3 +68,33 @@ def test_nodejs_attach_maps_from_container(
     )
     collapsed = parse_one_collapsed(collapsed_text)
     assert_collapsed(collapsed)
+
+
+@pytest.mark.parametrize("in_container", [True])
+@pytest.mark.parametrize("profiler_type", ["attach-maps"])
+@pytest.mark.parametrize("runtime", ["nodejs"])
+@pytest.mark.parametrize("application_image_tag", ["without-flags"])
+def test_twoprocesses_nodejs_attach_maps(
+    tmp_path: Path,
+    assert_collapsed: AssertInCollapsed,
+    profiler_type: str,
+    profiler_flags: List[str],
+    application_factory: Callable[[], _GeneratorContextManager],
+) -> None:
+    with application_factory() as pid1:
+        with application_factory() as pid2:
+            with SystemProfiler(
+                1000,
+                6,
+                Event(),
+                str(tmp_path),
+                False,
+                perf_mode="fp",
+                perf_inject=False,
+                perf_dwarf_stack_size=0,
+                perf_node_attach=True,
+            ) as profiler:
+                process1_collapsed = snapshot_pid_collapsed(profiler, pid1)
```

Comment:
Here, instead of invoking the profiler twice, you could run `results = profiler.snapshot()` once, and then get `results[pid1]` and `results[pid2]` from the results dict